Library/Homebrew/extend/os/linux/diagnostic.rb

```diff
@@ -143,13 +143,24 @@ def check_gcc_dependent_linkage
           return if gcc_dependents.empty?
 
           badly_linked = gcc_dependents.select do |dependent|
-            keg = Keg.new(dependent.prefix)
```

Comment:
I think the original code will work fine if you just replace
```ruby
dependent.prefix
```
here with
```ruby
dependent.any_installed_prefix
```
If you want to be sure, you can also try
```ruby
next unless dependent.any_installed_prefix.directory?
```